webpack.config.js

```diff
@@ -0,0 +1,71 @@
+import path from 'path';
+import webpack from 'webpack';
+import UglifyJSPlugin from 'uglifyjs-webpack-plugin';
+
+import {settings} from './gulp/config';
+
+const dev = {
+	entry: {
+		polyfills: './src/app/polyfills.js',
+		app: './src/app/index.js',
+		vendor: settings.sources.externalJs
```

Comment:
I guess if this isn’t used in any files in `/gulp` we shouldn’t store it in `gulp/config` and read it from package.json `bundleExternalJS` directly. Right?

Reply:
We do already use the 'gulp/config' file inside the webpack config for our dev entry. 

I thought we use the Baumeister.json only to let the user define settings, and the gulp/config js is used as our central settings store

Reply:
mhmm. webpack is triggered via gulp. This is an argument to handle it that way.

But: It felt a weird to me to use settings from `gulp/config` outside the `gulp` dir.
Don’t you think so?

Reply:
We could move our settings file to a more top level dir maybe?

Reply:
Thought about that. But I don’t wanna clutter up the root directory.

Don’t you feel comfortable with?
- only storing properties from `baumeister.json` in `gulp/config` which are needed in Gulp tasks
- using properties from `baumeister.json` directly in `webpack.config.js`

Reply:
@nicolaisueper ⬆️

Reply:
I thought about default values:

Maybe we want to decide to open more options to the user like `htmlSources` (especially if we refactored to a node module). That results in a new if / ternary at the source where it's needed (see [here](https://github.com/micromata/Baumeister/pull/211/commits/9ce1548e157b0a04ef19f5d7ed3174e9de677a37#diff-11e9f7f953edc64ba14b0cc350ae7b9dR57). If I haven't done [this](https://github.com/micromata/Baumeister/pull/211/commits/f6bd19900cd5e9cb7a4a8ac88eb0e769191031bc#diff-eddbfca195b18ab33db4917c1936198cR54) we would have the check at the `webpack.config.js`).
I think this makes the source less readable, but please tell me your opinion on this. :)

Reply:
And I see, I missed to set it to 'dev' in the else part 😆 fixing this now

Reply:
> I think this makes the source less readable, but please tell me your opinion on this. :)

I makes the webpack config more readable but also introduces some »magic« hidden in `gulp/config` between having a `dev` and `prod` property of `DefinePlugin` in `baumeister.json` and using just one `DefinePlugin` setting in the webpack config.

So we have three files which involved:
1. Define settings
2. Alter settings
3. Consume settings

Instead of just two:
1. Define settings
2. Consume settings

I personally would prefer not using a ternary for that all and instead using the appropriate setting in the `prod` and `dev` setting of `webpack.config.js`. 

Yes, this looks like code duplication and might be less clever, but it’s easier to comprehend.

Reply:
Alright, I'm going to refactor this tomorrow.